src/roompanel/RoomWidget.cpp

```diff
@@ -0,0 +1,252 @@
+    switch (static_cast<ColumnTypeEnum>(column)) {
+    case ColumnTypeEnum::NAME:
+        i = RoomMob::Field::NAME;
+        break;
+    case ColumnTypeEnum::LABEL:
+        i = RoomMob::Field::LABELS;
+        break;
+    case ColumnTypeEnum::POSITION:
+        i = RoomMob::Field::POSITION;
+        break;
+    case ColumnTypeEnum::EFFECTS:
+        i = RoomMob::Field::FLAGS;
+        break;
+    case ColumnTypeEnum::WEAPON:
+        i = RoomMob::Field::WEAPON;
+        break;
+    case ColumnTypeEnum::FIGHTING:
+        i = RoomMob::Field::FIGHTING;
+        break;
+    case ColumnTypeEnum::MOUNT:
+        i = RoomMob::Field::MOUNT;
+        break;
+    default:
+        return empty;
+    }
+    QVariant &variant = mob->field(i);
+    if (static_cast<QMetaType::Type>(variant.type()) != QMetaType::UInt) {
+        return variant;
+    }
+
+    // field contains the ID of another mob: try to resolve it
+    const auto iter = m_mobsById.find(variant.toUInt());
+    if (iter != m_mobsById.end()) {
+        const SharedRoomMob mob2 = iter->second;
+        if (mob2) {
+            return mob2->field(RoomMob::Field::NAME);
+        }
+    }
+    if (false) {
+        qDebug() << "mob id" << variant << "not found";
+    }
+    // failed to resolve ID of another mob:
+    // clear the ID before MUME reuses it
+    variant = empty;
+    return empty;
+}
+
+QColor RoomModel::getMobColor(const int row) const
+{
+    SharedRoomMob mob = getMob(row);
+    if (mob) {
+        return mob->color();
+    }
+    return QColor{};
+}
+
+bool RoomModel::isFightingYOU(const int row, const int column) const
+{
+    if (static_cast<ColumnTypeEnum>(column) == ColumnTypeEnum::FIGHTING //
+        && row >= 0 && size_t(row) < m_mobVector.size()) {
+        //
+        const SharedRoomMob mob = m_mobVector.at(size_t(row));
+        return mob && mob->field(RoomMob::Field::FIGHTING) == RoomMob::YOU;
+    }
+    return false;
+}
+
+void RoomModel::update()
+{
+    beginResetModel();
+    m_room.listMobs(m_mobsById, m_mobVector);
+    endResetModel();
+}
+
+// ------------------------------- RoomWidget ----------------------------------
+RoomWidget::RoomWidget(RoomManager *rm, QWidget *const parent)
+    : QWidget{parent}
+    , m_model{nullptr}
+    , m_table{nullptr}
+    , m_roomManager(rm)
+{
+    auto *layout = new QVBoxLayout(this);
+    layout->setAlignment(Qt::AlignTop);
+    layout->setContentsMargins(0, 0, 0, 0);
+    layout->setSpacing(0);
+
+    m_model = std::make_unique<RoomModel>(this, *rm->getRoom());
+
+    m_table = std::make_unique<QTableView>(this);
+    m_table->setSelectionMode(QAbstractItemView::ContiguousSelection);
+    m_table->setSelectionBehavior(QAbstractItemView::SelectRows);
+    m_table->horizontalHeader()->setStretchLastSection(false);
+    m_table->horizontalHeader()->setSectionResizeMode(QHeaderView::Interactive);
+    m_table->setModel(m_model.get());
+    /// TODO: [Cosmos] m_table->setItemDelegate(new RoomDelegate(this));
```

Comment:
Care to make this comment be more human readable?

Reply:
This is a leftover of copy-pasting `GroupPanel` code. To be honest, I have no idea what's an `ItemDelegate` is. Any reference?

Reply:
QStyledItemDelegate allows you to replace the text with some icon: https://doc.qt.io/qt-5/qtwidgets-itemviews-stardelegate-example.html

We used it in the GroupPanel to change the text to little icons

Reply:
I see. Removed, not using icons in RoomPanel at the moment.